src/Analysis/Ast/Test/PathClassificationTests.cs

```diff
@@ -0,0 +1,189 @@
+                new PythonLibraryPath(venvSitePackages, PythonLibraryPathType.Site),
+            };
+
+            var fromUser = new[] {
+                "./src",
+            };
+
+            var (interpreterPaths, userPaths) = PythonLibraryPath.ClassifyPaths(root, _fs, fromInterpreter, fromUser);
+
+            interpreterPaths.Should().BeEquivalentToWithStrictOrdering(new[] {
+                new PythonLibraryPath(venvLib, PythonLibraryPathType.StdLib),
+                new PythonLibraryPath(venv, PythonLibraryPathType.StdLib),
+                new PythonLibraryPath(venvSitePackages, PythonLibraryPathType.Site),
+            });
+
+            userPaths.Should().BeEquivalentToWithStrictOrdering(new[] {
+                new PythonLibraryPath(src, PythonLibraryPathType.Unspecified),
+            });
+        }
+
+        [TestMethod]
+        public void NormalizeUser() {
+            var appPath = TestData.GetTestSpecificPath("app.py");
+            var root = Path.GetDirectoryName(appPath);
+
+            var src = Path.Combine(root, "src");
+
+            var fromUser = new[] {
+                "./src/",
+            };
+
+            var (interpreterPaths, userPaths) = PythonLibraryPath.ClassifyPaths(root, _fs, Array.Empty<PythonLibraryPath>(), fromUser);
+
+            interpreterPaths.Should().BeEmpty();
+
+            userPaths.Should().BeEquivalentToWithStrictOrdering(new[] {
+                new PythonLibraryPath(src, PythonLibraryPathType.Unspecified),
+            });
+        }
+
+        [TestMethod]
+        public void NestedUser() {
+            var appPath = TestData.GetTestSpecificPath("app.py");
+            var root = Path.GetDirectoryName(appPath);
+
+            var src = Path.Combine(root, "src");
+            var srcSomething = Path.Combine(src, "something");
+            var srcFoo = Path.Combine(src, "foo");
+            var srcFooBar = Path.Combine(srcFoo, "bar");
+
+            var fromUser = new[] {
+                "./src",
+                "./src/something",
+                "./src/foo/bar",
+                "./src/foo",
+            };
+
+            var (interpreterPaths, userPaths) = PythonLibraryPath.ClassifyPaths(root, _fs, Array.Empty<PythonLibraryPath>(), fromUser);
+
+            interpreterPaths.Should().BeEmpty();
+
+            userPaths.Should().BeEquivalentToWithStrictOrdering(new[] {
+                new PythonLibraryPath(src, PythonLibraryPathType.Unspecified),
+                new PythonLibraryPath(srcSomething, PythonLibraryPathType.Unspecified),
+                new PythonLibraryPath(srcFooBar, PythonLibraryPathType.Unspecified),
+                new PythonLibraryPath(srcFoo, PythonLibraryPathType.Unspecified),
+            });
+        }
+
+        [TestMethod]
+        public void InsideStdLib() {
+            var appPath = TestData.GetTestSpecificPath("app.py");
+            var root = Path.GetDirectoryName(appPath);
```

Comment:
Do we have test where user paths and interpreter paths somehow intersect?

Reply:
I'm not sure if that condition is possible. The only way for a path to become an interpreter path is to either be within a stdlib directory or to be outside the workspace. There couldn't be a user path inside an interpreter path without that path also being an interpreter path by the first condition.

In the future, I'd like to offer more control over this (and have two lists which explicitly force a path into one of the two categories), but at the moment I don't think it's a behavior that can happen.

Reply:
Ok